Remove Kafka meters on unregistration notification

izeye opened the pull request:
This PR removes Kafka meters on unregistration notification.

Closes gh-1743

shakuzen commented:
Thank you for the quick turnaround on a fix for this. I'd like to make sure we have a test for this. Would you be willing to add one? Otherwise, I can work on that.

izeye commented:
@shakuzen Thanks for the feedback! I added tests and handled a missed case thanks to the tests. I also changed the base branch to 1.1.x as it was master accidentally.

shakuzen commented:
One thing that came to mind is that if Kafka consumers are created and closed before a full step interval, the metrics from it may never be published. I'm not sure that's even an issue we should concern ourselves with, but I don't think we have any global metrics that would show this behavior happening currently. This is one of the things we can hope to fix with the Kafka metrics overhaul. In the meantime, we want this fix so we're not causing memory issues.
